storages/python/python-storage-plugins/mongodb.py

```diff
@@ -0,0 +1,94 @@
+import json
+import sys
+from urllib.parse import quote_plus, urlparse
+
+import pymongo
+
+import dlite
+from dlite.options import Options
+from dlite.utils import instance_from_dict
+
+
+class mongodb(dlite.DLiteStorageBase):
+    """DLite storage plugin for PostgreSQL."""
+    def open(self, uri, options=None):
+        """Opens `uri`.
+
+        The `options` argument provies additional input to the driver.
+        Which options that are supported varies between the plugins.  It
+        should be a valid URL query string of the form:
+
+            key1=value1;key2=value2...
+
+        An ampersand (&) may be used instead of the semicolon (;).
+
+        Options:
+        - database: Name of database to use (default: "dlite")
+        - collection: Name of collection to use (default: "dlite_coll")
+        - user: User name.
+        - password: Password.
+        - mode: "r" for opening in read-only mode, "w" for read/write mode.
+        - authMechanism: Authentication mechanism (default: "SCRAM-SHA-256")
+        - mock: Whether to use mongomock.
+        """
+        self.options = Options(
+            options,
+            defaults='database=dlite;collection=dlite_coll;mode=w;'
+            'authMechanism=SCRAM-SHA-256;mock=no',
+        )
+        opts = self.options
```

Comment:
Is there a reason you are redefining the `self.options` attribute locally here? It seems you could do without this and simply use `self.options` anywhere you are currently using the less clear `opts` variable?

Reply:
Only reason id to make the code easier to read by reducing line lengths.
Moved `self.options = opts` to the end of the function.